schema.php

```diff
@@ -13,18 +13,35 @@
 	global $wpdb;
 	foreach ( $queries as $k => $q ) {
 		// Replace meta_key index with one that indexes meta_value as well
-		if ( preg_match( '|CREATE TABLE ([^ ]*)|', $q, $matches ) && $wpdb->postmeta === $matches[1] ) {
-			$queries[ $k ] = str_replace(
-				'KEY meta_key (meta_key(191))',
-				sprintf( 'KEY %s', get_postmeta_key_value_index() ),
-				$q
-			);
+		if ( false !== strpos( $q, "CREATE TABLE {$wpdb->postmeta}" ) ) {
+			$search = 'KEY meta_key (meta_key(191))';
+			if ( false !== strpos( $q, $search ) ) {
+				$status = _get_table_status( $wpdb->postmeta );
+				// For InnoDB, Data_length is the approximate amount of space allocated for the clustered index, in bytes.
+				// Each InnoDB table has a special index called the clustered index that stores row data.
+				if ( isset( $status['Data_length'] ) && $status['Data_length'] < 52428800 ) {
```

Comment:
52428800 is 50 MiB; we can discuss this limit.

On a Intel(R) Xeon(R) E-2136 CPU @ 3.30GHz physical server, it took 0.68 seconds for MySQL to create this index (data length was 55,132,160 bytes).

For site 2160, it took 1 min 18.038 sec to build the index (data length was 744,488,960 bytes)

Reply:
By dividing 744M by 78, we get roughly 9,544,730 bytes/sec; so maybe we need to lower the limit to 10 MiB.